### PR TITLE
feat: pre call diagnostics web doc

### PR DESCRIPTION
**Pull request by eswarclynn**

### CODE REVIEW

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+If you want a list of devices to choose from before testing your devices you'll need to request corresponding permissions using
+
+```js
+diagnostics?.requestPermission({ audio: true, video: true }).catch((error) => setError(error));
+```
+
+## Test microphone
+
+To test if the microphone is correctly working, call the `startMicCheck` API passing preffered check duration as `timeInMillis` parameter. Diagnostics SDK will record the microphone input into a temporary file.
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:34]: Spelling of "preferred"

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
+---
+title: Pre Call Diagnostics
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:2]: Pre-call Diagnostics

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+nav: 11.4
+---
+
+## Pre Call Diagnostic APIs
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:6]: Hyphen between pre & call

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+element.src = hmsDiagnostics?.getRecordedAudio() || 'https://100ms.live/test-audio.wav';
+if (typeof element.setSinkId !== 'undefined') {
+    // @ts-ignore
+    const { audioOutputDeviceId } = hmsStore.getState(selectLocalMediaSettings);
+    element.setSinkId(audioOutputDeviceId);
+}
+element.play();
+```
+
+You can set the audio output device ID using HMSActions like `hmsActions.setAudioOutputDevice(deviceId);`
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:64]: `HMSActions`

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+## Request Permissions
+
+If you want a list of devices to choose from before testing your devices you'll need to request corresponding permissions using
+
+```js
+diagnostics?.requestPermission({ audio: true, video: true }).catch((error) => setError(error));
+```
+
+## Test microphone
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:32]: Test Microphone

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    inputDevice,
+    onError:,
+    onStop,
+    time, // in ms, default 10 seconds
+});
+```
+
+You can call `stopMicCheck` if you need to interrupt the check before the specified time has passed.
+
+## Test speaker
```
Comment (mntrspace) [inline docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:49]: Test Speaker

**Review by mntrspace — CHANGES_REQUESTED** [review]

File: docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx (hunk 4 above)
Comment (eswarclynn) [reply docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:64]: The JS convention is lowercased-camelcase

Comment (mntrspace) [reply docs/javascript/v2/how-to-guides/debugging/pre-call-diagnostics.mdx:64]: Oh I meant that HMSActions wasn't enclosed in code quote in the sentence.

**Review by mntrspace — APPROVED** [review]

**Pull request merged**